### ISSUE TITLE
caogao

### ISSUE THREAD

**Issue by irfanICMLL** [irfanICMLL/colorful-_cartoons#1]
can i leave a comment on it?

**Comment by irfanICMLL** [comment]
wonderful!

**Comment by irfanICMLL** [comment]
We quantify these observations using popularity index. For the ith picture generate by the jth algorithm, the popularity ij is define as:
popularity_ij=log(nlike_ij+c/ndislike_ij+c)
where i=40, j=4, c=1 in our test and nlike_ij(ndislike_ij) represent for the number of subjects who choose "like(dislike)" on this picture_ij. c is set for smoothness  
By this way, the popularity of algorithom j is:
popularity_j=log(sum(nlike_ij)+c/sum(ndislike_ij)+c)
From the table 1 we can see that auto-painter is the most popular algorithm among all candidates while pix2pix is the most unpopular ones. We also test the effect of removing feature loss (labeled as tv loss). This variant results in unstable performance as it has the largest variance, suggesting that under the condition of the different input, it produced different quality results. Adding feature loss improves the ability of the system to adapt for diversity of sketches. However, adding a tv loss also improve the popularity and decrease the variance. Combining all terms, auto-painter obtains best performance.

User-guided Colorization
Given an input image of resolution 512x512, the auto-painter can generate colorful results within 1 second, which enables instant feedback to design an interactive image editing tools. The auto-painter trained on the minions provides a tool for users to design virtual images of which respects “minions’ style”. As it shown in figure X, based on the initial sketch, one can changes the contour lines, the glass or the gesture, etc. It is interesting that even on the simple hand draw sketches, the auto-painter can result in “minions’ style” images (see figure X+1). 


Conclusions
In this paper, we proposed a system called auto-painter which can transfer sketches to colorful images. Our approach is based on ‘U-net’ and we employ stronger constraints in the system. Contrast test and perception experiments were made to analysis the effect of different part in the loss function, and the results show that the auto-painter can generate high-quality images stably. We also trained the auto-painter to adapt to the different color control inputs. With color blocks, our network can adapt the synthesis results to satisfy different color constraints. Besides, we developed two datasets of high-quality cartoon images with their sketches and defined statistics and test method to evaluate the synthesis images.
Despite the promising results, our current system suffers from the difficulties of adjusting parameters just like other deep learning model. Complex network structure results in low training speed. For example, on our Girl datasets, it takes 2~3 days to get convincing results with a single Tesla K80 GPU. Fortunately, the feed-forward structure enables a practical test time under 1 second.

**Comment by irfanICMLL** [comment]
Abstract
Synthesizing cartoon images from simple sketches is both interesting and practical. But it is a clearly underconstrained problem. We develop a deep neural network called auto-painter to deal with this problem. With a conditional advertising training and strong constraints between the sketch and images, the network learns a mapping from the input sketch and the colorful images. We also train the model to adapt to the color block control. Convincing results show that the auto-painter performs better that existing image-to-image method. The network is feed-forward which enables users to see the results of their edits in real time.

**Comment by irfanICMLL** [comment]
Color controlled sketch synthesis 
When transfer a gray scale sketch into a real photo, the color scheme is relatively fixed. However, the cartoon images are full of imagination and contain manifold color matches. With simple black and white sketch as input causes the problem to be underconstrained, since one sketch can correspond to images colored in many different ways. Besides, users may prefer to see different color matches instead of a fixed one. In this section, we discuss how to add color control to the auto-painter. The structure of the network remained unchanged so does the loss function. The training set is replaced Japanimation to get more color information. Follow the description in Section 3.1, we generate sketches with color control blocks of Japanimation.
Figure X+2 shows the results of reconstructing Japanimation based on synthesis color control blocks. We can see that for the complicated task, the auto-painter identify the edges of the images successfully and make feasible color schemes based on the color block. The color of the control block is rendered around in a particular region naturally. We also test the model with user input and the colors of the blocks deviate a lot from the colors in the ground-truth image (see figure X+3). Nevertheless, the auto-painter is able to paint the color in object boundaries and distinguish different parts of the sketch.

**Comment by irfanICMLL** [comment]
Hi, taizan
I am the author of the Auto-Painter. I feel so sorry that I didn’t notice your work before I finish my article. 
The auto-painter is done when I was an intern in Samsung. They give me a paper called “scribbler: Controlling Deep Image Synthesis with Sketch and Color”, and told me to do some related researches about the image generation. Image generation is quit an interesting field, and there may be a lot of applications. Samsung has been committed to the research of colorization, but before the coming boom of artificial intelligence, they use some graphics methods. So they want me to do some research about painting a sketch to an image with GAN. I investigate about related articles and find the pix2pix project. Both the pix2pix and the scribbler focus on the generating of real pictures, such as faces, maps and bedrooms. They don’t refer to the cartoon image generations. So I start my experiments. 
The sketch synthesis algorithm is inspired by the project on the github: XDoG in matlab. I rewrite it in python as a filter to produce the training set. The training set of Minions was collected from the Baidu Images with a simple crawler I write myself. The training set of Japanimation was collected by my colleagues in Samsung on a cartoon website with a more complex crawler.
I follow the basic method from scribble to add new constrain on the pix2pix projects and find the results improved significantly. Both the pix2pix model and the auto-painter model were trained with the same training set and shared the same parameters. The only difference between them was the objective function. I didn’t make unfair comparison between them. 
I think the addition parts of the objective function really work, and network can be used in such an interesting field, so I write an article to share with others. I feel so sorry that I only concentrate on the published articles and ignore your project. 
After I publish my paper on the arXiv, someone forward it to Chinese WeiBo. I read about the comments and know that there is a project called PainterChanier. I search for the article but find nothing. Then I search on the github, and finally I find you on the github. 
So I write such a long letter to explain all the stories. This is the first time that I publish an article to the arXiv. I haven’t expect to cause such a big misunderstanding.

Best regards,
Irfan
